### PR TITLE
Proto2 primitive type conversion fix.

### PR DESCRIPTION
**Pull request by TristonianJones**
Closes issue #149 by adding appropriate `NativeToValue` conversions for `proto2` primitive fields.
In `proto2` primitive fields are generated as pointers to primitive go types, whereas in `proto3` the
simple type is used, e.g. `string` (proto3) vs. `*string` (proto2).

### CODE REVIEW

File: interpreter/interpreter_test.go [hunk 1]
```diff
@@ -389,27 +391,128 @@ func TestInterpreter_BuildObject(t *testing.T) {
 	}
 }
 
+func TestInterpreter_GetProto2PrimitiveFields(t *testing.T) {
+	src := common.NewTextSource(`
+	a.single_int32 == a.single_int64 &&
```
Comment (madi8229) [inline interpreter/interpreter_test.go:398]: Do we have any comments to explain why this works while CEL does not support type conversion? Thank you.

File: interpreter/interpreter_test.go [hunk 2]
```diff
@@ -389,27 +391,128 @@ func TestInterpreter_BuildObject(t *testing.T) {
...
+		SingleDouble: &six,
+		SingleString: &str,
+		SingleBool:   &truth,
+	}
+	result, state := eval.Eval(NewActivation(map[string]interface{}{
+		"input": input,
+	}))
+	got, ok := result.(ref.Value).Value().(bool)
+	if !ok {
+		t.Fatalf("cannot cast result to int64: result=%v state=%v", result, state)
```
Comment (madi8229) [inline interpreter/interpreter_test.go]: We have several fields here, why we know if it is not ok, this is the reason? Thank you.

File: interpreter/interpreter_test.go (hunk 1 above)
Comment (TristonianJones) [reply interpreter/interpreter_test.go:398]: In CEL, 32-bit proto types are widened to 64-bits, so even though the implicit type conversion would be an error normally, the widening happens at the object layer and CEL doesn't consider them different types.

File: interpreter/interpreter_test.go (hunk 2 above)
Comment (TristonianJones) [reply interpreter/interpreter_test.go]: Comment updated to match the actual conditions under test.

**Review by madi8229 — APPROVED** [review]
LGTM

**Comment by TristonianJones** [comment]
Closes #149

**Pull request merged**